### ISSUE TITLE
CollisionSegment mixes up SurfacePoint & InteriorPoint

### ISSUE THREAD

**Issue by jeroendenhaan** [panda3d/panda3d#974]
Panda version: 1.10.6

Setting up a CollisionSegment as part a line-of-sight has shown strange behavior; it seems the first object hit gets its SurfacePoint on exit of the into-object, and the so-called InteriorPoint is shown upon entry of the first object. CollisionRay works as expected (no InteriorPoint is shown in the visualizer here, but SurfacePoint is correct). Screenshot shows behavior of CollisionSegment. PointA of segment is in bottom left, PointB is upper right. Furthermore, switching PointA and PointB doesn't seem to change the behavior.

Discussed in forum here: https://discourse.panda3d.org/t/collisions-surfacepoint-and-interiorpoint-strange-behavior/26254/6

Tried:
- swithcing originating model for an empty nodepath (no difference)
- not scaling collision into-objects (no difference)
- CollisionRay works as expected
- only tested with CollisionBoxes

![coll](https://user-images.githubusercontent.com/2821356/87818818-8401fc80-c86b-11ea-9d1a-424412bcb0fe.png)

**Comment by rdb** [comment]
I think it's picking the closest point of entry.  Would you rather expect that the surface point is always the point in the target volume that is closest to point A and the interior point is the point in the target volume that is closest to B?

**Comment by sabvdf** [comment]
> I think it's picking the closest point of entry (irrespective of the direction of the segment). Would you rather expect that the surface point is always the point in the target volume that is closest to point A and the interior point is the point in the target volume that is closest to B?

Yes, I've just spent quite some time breaking my heading over what was happening with my segment: this is the expected behavior based on both of these sentences that are written in the [CollisionSegment documentation](https://docs.panda3d.org/1.10/python/reference/panda3d.core.CollisionSegment):

> It’s similar to a CollisionRay, except it does not continue to infinity.

This would mean it is directed.

> It does have an ordering, from point A to point B. If more than a single point of the segment is intersecting a solid, the reported intersection point is generally the closest on the segment to point A.

Except for the word 'generally', this also implies it returns Surface versus Interior Points in the direction of A->B.

Right now, I'm choosing to do a comparison every frame of `entry.getSurfacePoint(self.cam).y > entry.getInteriorPoint(self.cam).y` (the segment being a child of `self.cam`)
to avoid raycasting to infinity which would be a lot worse with the amount of colliders in my scene, but it would be nicer if the CollisionSegment would behave as described.
